hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/bucket/OMBucketCreateRequest.java

```diff
@@ -149,7 +149,9 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
     } else {
       omBucketInfo = OmBucketInfo.getFromProtobuf(bucketInfo);
     }
-
+    if (omBucketInfo.getBucketLayout().isFileSystemOptimized()) {
+      omMetrics.incNumFSOBucketCreates();
+    }
```

Comment:
The null check is added only for testing purposes. It will never be null in real-world scenarios. We could however move the metrics-related code into the else block in case we are running into any NPEs.

Reply:
@ayushtkn @JyotinderSingh 

```
 if (bucketInfo.getBucketLayout() == null)
       // sets to defaultValue. Here it can be OBS or FSO.

 } else {
      omBucketInfo = OmBucketInfo.getFromProtobuf(bucketInfo);
 }
```

Since there is already an explicit `null` check [OMBucketCreateRequest.java#L142](https://github.com/apache/ozone/blob/master/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/bucket/OMBucketCreateRequest.java#L142) to safeguard NPE, I think it won't be null in any case and additional null check is not required, isn't it?

Reply:
Hey Rakesh,
Got little confused, you mean to say:
```
    if (bucketInfo.getBucketLayout() == null || bucketInfo.getBucketLayout()
        .equals(BucketLayoutProto.LEGACY)) {
      // Bucket Layout argument was not passed during bucket creation.
      String omDefaultBucketLayout = ozoneManager.getOMDefaultBucketLayout();
      BucketLayout defaultType = BucketLayout.fromString(omDefaultBucketLayout);
      omBucketInfo = OmBucketInfo.getFromProtobuf(bucketInfo, defaultType);
    } else {
      omBucketInfo = OmBucketInfo.getFromProtobuf(bucketInfo);
    }
```
Post this block there won't be any ``null`` check required because ``omBucketInfo`` is being set to some value in this block?
Then in that case value is being set to ``omBucketInfo`` and in the new code added there is an operation ``omBucketInfo.getBucketLayout().isFileSystemOptimized()``, we didn't modify ``omBucketInfo.getBucketLayout()``? So if that is ``null`` and we call for ``isFileSystemOptimized()`` on top of it, this would throw NPE, right?

Or if you mean to say pull the new code in the else block, yeps that won't require any null check.

Just had a quick look on it, might have messed up. Please don't hold this for me. You can proceed if you are convinced with the present code. :-)

Reply:
Hey Ayush,
```java
BucketLayout defaultType = BucketLayout.fromString(omDefaultBucketLayout);
omBucketInfo = OmBucketInfo.getFromProtobuf(bucketInfo, defaultType);
```
These lines inside the `if` condition will set the bucketLayout for omBucketInfo in case it is not present - so it will not be null. 
The second argument in `OmBucketInfo.getFromProtobuf` call is the bucket layout being passed.

Reply:
Cool